clients/src/main/java/org/apache/kafka/common/requests/ProduceResponse.java

```diff
@@ -267,7 +267,7 @@ protected Struct toStruct(short version) {
                         .set(ERROR_CODE, errorCode)
                         .set(BASE_OFFSET_KEY_NAME, part.baseOffset);
                 if (partStruct.hasField(LOG_APPEND_TIME_KEY_NAME))
-                    partStruct.set(LOG_APPEND_TIME_KEY_NAME, part.logAppendTime);
+                    partStruct.setIfExists(LOG_APPEND_TIME_KEY_NAME, part.logAppendTime);
```

Comment:
The if condition is not needed anymore right?

Reply:
ack